multi-tenancy-observability/README.md

```diff
@@ -0,0 +1,270 @@
+
+We are authenticated on Grafana through a JWT token provided by teleport.
+But that token is not forwarded to the datasource as this is not using the Grafana OAuth plugin to authenticate. Due to this, we cannot access the users groups in the grafana-multi-tenant-proxy.
+
+We are considering a number of options that are not ideal at the moment:
+
+- having double datasource: one for customer with OAuth authentication and another dedicated to GiantSwarm people authenticated throw teleport.
+- having double grafana (same idea than above but for the all grafana instance, not just the datasource).
+- share an authentication secret via cookies.
+- other ideas are welcome!
+
+#### Mapping user's groups to tenants
+
+What interface could we offer our customers to enable them to define the mapping between their groups and their tenants:
+
+- Custom resources like TenantCR, GroupCR
+- UI in happa
+- other ideas ?
+
+## Write path
+
+### Current architecture for the write path
+
+On the write path, we currently use promtail to collect container and machine logs and grafana agent to collect kubernetes events.
+
+We currently need those 2 tools because:
+1. Promtail cannot get kubernetes events
+2. Grafana Agent is going through quite a big rewrite in it's configuration and all features we needed were not supported when we deployed it
+
+Here is what the flow looks like:
+1. The logging-operator creates a `password` for the cluster's logging agents to be able to send logs to Loki and configures them (actual configuration of what to scrape and so on) via an extra-config.
+2. The logging-operator adds the new cluster_id/password pair to the proxy configuration
+3. The logging agents collect and send logs to the loki-multi-tenant-proxy which validates the basic authentication sent in the http header and transform it into a tenant information (X-Scope_OrgID)
+4. Loki writes the data to storage for the configured tenant (cluster_id in that case)
+
+### Configuring multi-tenancy on the write path
+
+#### Different tenant configuration options
+
+After a few different customer calls, we found out that they have different needs when it comes to multi-tenancy.
+
+##### Cluster == Tenant
+
+This is the approach we are currently using for technical reasons. We do not think this would match with customers workload as they mostly have a lot of customers sharing clusters.
+
+##### Namespace == Tenant
+
+This approach makes it really easy for us and customers to configure tenants and it would definitely be easy to onboard teams (e.g. adding a label on the namespace would make the agents collect logs from the containers in the namespace).
+
+But this approach brings a few questions:
+1. What tenancy should common internal apps use (e.g. prometheus-operator, api-server, cilium logs are accessed by multiple internal teams)? Do we need to ensure customers are putting those apps into a namespace accessible by all?
```

Comment:
For my discussion with customers, prometheus-operator was moved to another namespace because teams did not have access to kube-system logs :D

Reply:
So, my main questions in this case is more for our managed apps because customers can install them in namespaces they want and it could also be running with their own workload like it is the case with prometheus-operator. Now we technically have access to all the logs because we manage the clusters but in loki, we could theoretically reduce our access to only kube-system, giantswarm namespaces and managed apps logs only.

Do we want to go in that direction or should Giant Swarm have access to all logs by default?